src/main/java/de/tum/in/www1/artemis/service/exam/ExamAccessService.java

```diff
@@ -226,24 +153,20 @@ public ExamAccessService(ExamRepository examRepository, ExerciseGroupRepository
      * @param courseId        The id of the course
      * @param examId          The id of the exam
      * @param exerciseGroup   The exercise group
-     * @param <X>             The type of the return type of the requesting route so that the
-     *                        response can be returned there
-     * @return an Optional with a typed ResponseEntity. If it is empty all checks passed
      */
-    public <X> Optional<ResponseEntity<X>> checkCourseAndExamAndExerciseGroupAccess(Role role, Long courseId, Long examId, ExerciseGroup exerciseGroup) {
-        Optional<ResponseEntity<X>> courseAndExamAccessFailure = switch (role) {
-            case INSTRUCTOR -> checkCourseAndExamAccessForInstructor(courseId, examId);
-            case EDITOR -> checkCourseAndExamAccessForEditor(courseId, examId);
-            default -> Optional.of(forbidden());
-        };
-        if (courseAndExamAccessFailure.isPresent()) {
-            return courseAndExamAccessFailure;
+    public void checkCourseAndExamAndExerciseGroupAccessForRoleElseThrow(Role role, Long courseId, Long examId, ExerciseGroup exerciseGroup) {
+        switch (role) {
+            case ADMIN -> checkCourseAndExamAccessForRoleElseThrow(Role.ADMIN, courseId, examId);
+            case INSTRUCTOR -> checkCourseAndExamAccessForRoleElseThrow(Role.INSTRUCTOR, courseId, examId);
+            case EDITOR -> checkCourseAndExamAccessForRoleElseThrow(Role.EDITOR, courseId, examId);
+            case TEACHING_ASSISTANT -> checkCourseAndExamAccessForRoleElseThrow(Role.TEACHING_ASSISTANT, courseId, examId);
+            case STUDENT -> checkCourseAndExamAccessForRoleElseThrow(Role.STUDENT, courseId, examId);
+            case ANONYMOUS -> checkCourseAndExamAccessForRoleElseThrow(Role.ANONYMOUS, courseId, examId);
         }
```

Comment:
Same idea as the other switch: Why can’t this be simplified to a single `checkCourseAndExamAccessForRoleElseThrow(role, courseId, examId);`?